src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,131 @@
+                throw new DukeException("Task cannot be found within the task list! Please fix your machigai!");
+            }
+
+            switch (commandType) {
+            case DELETE:
+                return new Delete(taskId);
+            case UNMARK:
+                return new Unmark(taskId);
+            case MARK:
+                return new Mark(taskId);
+            default:
+                throw new DukeException("Invalid Command Type!");
+            }
+
+        } catch (NumberFormatException e) {
+            throw new DukeException("Task ID has to be an integer!");
+        }
+
+    }
+
+    /**
+     * Method that parses the string input and performs the correct action
+     * @param input user input into the command lines
+     * @return true if the command that has been inputted equals "bye", else return false
+     * @throws DukeException if the command is invalid
+     */
+    public Command parseInput(String input) throws DukeException {
+        String[] inputArray = input.split(" ");
+
+        // single command
+        if (inputArray.length == 1) {
+            if (inputArray[0].equalsIgnoreCase("list")) {
+                return new List();
+            } else if (inputArray[0].equalsIgnoreCase("bye")) {
+                return new Bye();
+            } else {
+                throw new DukeException("Sumimasen! I don't recognize this command. Please try again!");
+            }
+        }
+
+        // multi command
+        if (inputArray.length > 1) {
+            CommandType commandType = CommandType.getCommandType(inputArray[0]);
+            StringBuilder taskDetailsBuilder = new StringBuilder();
+
+            for (int i = 1; i < inputArray.length; i++) {
+                if (i != inputArray.length - 1) {
+                    taskDetailsBuilder.append(inputArray[i]).append(" ");
+                } else {
+                    taskDetailsBuilder.append(inputArray[i]);
+                }
+            }
+
+            // task variables
+            String taskDetails = taskDetailsBuilder.toString();
+            String description = "";
+            String date = "";
+            String dateTime = "";
+
+            if (taskDetails.contains("/by")) {
+                description = taskDetails.split("/by", 2)[0];
+                date = taskDetails.split("/by", 2)[1].substring(1);
+            } else if (taskDetails.contains("/at")) {
+                description = taskDetails.split("/at", 2)[0];
+                dateTime = taskDetails.split("/at", 2)[1].substring(1);
+            }
+
+            switch (commandType) {
+            case FIND:
+                return new Find(inputArray[1]);
+            case DELETE:
+                return taskAction(CommandType.DELETE, inputArray[1]);
+            case UNMARK:
+                return taskAction(CommandType.UNMARK, inputArray[1]);
+            case MARK:
+                return taskAction(CommandType.MARK, inputArray[1]);
```

Comment:
Perhaps you can consider making inputArray[1] a local variable in the method so as to give it a descriptive name for easy reference?